File: js/problem7.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    if ((x > -10) && (x < 10)) {
        return x;
    }
    
    let sign = 1;
    if (x < 0) {
        sign = -1;
    }
    
    x = Math.abs(x);
    
    let res = 0;
    
    while (x > 0) {
        let temp = x % 10;
        res = (res * 10) + temp;
        x = Math.floor(x / 10);
    }
    
    res *= sign;
    
    if ((res < -Math.pow(2, 31)) || (res > (Math.pow(2, 31) - 1))) {
        return 0;
    }
    
    return res;
};

File: leetcodeJava/src/main/java/Problem7.java
// 7. Reverse Integer

class Solution {
    public int reverse(int x) {
        if ((x > -10) && (x < 10)) {
            return x;
        }
        
        int res = 0;
        while (x != 0) {
            int temp = x % 10;
            if ((res > Integer.MAX_VALUE / 10) || ((res == Integer.MAX_VALUE / 10) && (temp > 7))) {
                return 0;
            }
            if ((res < Integer.MIN_VALUE / 10) || (res == Integer.MIN_VALUE / 10) && (temp < -8)) {
                return 0;
            }
            res = (res * 10) + temp;
            x /= 10;
        }
        
        return res;
    }
}